lib/web_ui/lib/src/engine/view_embedder/hot_restart_cache_handler.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../dom.dart';
+import '../safe_browser_api.dart';
+
+/// Handles elements that need to be cleared after a hot-restart.
+class HotRestartCacheHandler {
+  HotRestartCacheHandler() {
+    if (_elements.isNotEmpty) {
+      // We are in a post hot-restart world, clear the elements now.
+      _clearAllElements();
+    }
+  }
+
+  /// This is state persistent across hot restarts that indicates what
+  /// to clear.  Delay removal of old visible state to make the
+  /// transition appear smooth.
```

Comment:
I feel like the second sentence belongs somewhere else, since a "name" can't possibly delay anything.

Or perhaps we can reword this something like: "This variable stores state clean-up callbacks that will be called after the hot restart is complete. This makes the restart appear smooth because the old UI continues to be visible during the restart."

Reply:
Yes, I rewrote the docs of this class so they make more sense.